src/couch/src/couch_server.erl

```diff
@@ -77,16 +77,20 @@ open(DbName, Options0) ->
     case ets:lookup(couch_dbs, DbName) of
     [#entry{db = Db0, lock = Lock} = Entry] when Lock =/= locked ->
         update_lru(DbName, Entry#entry.db_options),
-        {ok, Db1} = couch_db:incref(Db0),
-        couch_db:set_user_ctx(Db1, Ctx);
+        case couch_db:incref(Db0) of
+            {down,Db0} -> open(DbName, Options0);
```

Comment:
You should add a Retry argument to this function so that we limit ourself to a defined number of attempts or else this could lead to infinite looping (granted that's unlikely but its a possibility if we don't prevent it).

Reply:
And we should also do the same retry limiting for create as well.